libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthHelper.cs

```diff
@@ -0,0 +1,423 @@
+            {
+                turnContext.TurnState.Add<object>(TurnStateConstants.OAuthLoginTimeoutKey, TimeSpan.FromMilliseconds(settings.Timeout.Value));
+            }
+
+            // Set input hint
+            if (string.IsNullOrEmpty(prompt.InputHint))
+            {
+                prompt.InputHint = InputHints.AcceptingInput;
+            }
+
+            await turnContext.SendActivityAsync(prompt, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Method to process Event and Invoke activities that are part of the Bot Framework Protocol oauth support.
+        /// </summary>
+        /// <param name="dc">The DialogContext.</param>
+        /// <param name="userTokenClient">The user token client.</param>
+        /// <param name="connectionName">The ConnectionName.</param>
+        /// <param name="cancellationToken">CancellationToken.</param>
+        /// <returns>A Task of DialogTurnResult.</returns>
+        public static Task<DialogTurnResult> OnContinueWithNonMessageActivityAsync(DialogContext dc, UserTokenClient userTokenClient, string connectionName, CancellationToken cancellationToken)
+        {
+            switch (dc.Context.Activity.Type)
+            {
+                case ActivityTypes.Event:
+                    return OnContinueEventActivityAsync(dc, cancellationToken);
+
+                case ActivityTypes.Invoke:
+                    return OnContinueInvokeActivityAsync(dc, userTokenClient, connectionName, cancellationToken);
+
+                default:
+                    // unexpected Activity type
+                    return Task.FromResult(Dialog.EndOfTurn);
+            }
+        }
+
+        /// <summary>
+        /// Initialize the persisted state of the prompt. This is used in Begin with the data being retrieved in Continue.
+        /// </summary>
+        /// <param name="timeout">The Timeout.</param>
+        /// <param name="dc">The DialogContext.</param>
+        /// <param name="options">Persisted options.</param>
+        public static void InitializeState(int? timeout, DialogContext dc, object options)
+        {
+            var state = dc.ActiveDialog.State;
+            state[PersistedOptions] = options;
+            state[PersistedState] = new Dictionary<string, object>
+            {
+                { "AttemptCount", 0 },
+            };
+
+            state[PersistedExpires] = DateTime.UtcNow.AddMilliseconds(timeout ?? (int)TurnStateConstants.OAuthLoginTimeoutValue.TotalMilliseconds);
+            state[PersistedCaller] = CreateCallerInfo(dc.Context);
+        }
+
+        /// <summary>
+        /// Check whether the persisted timout has now expired. This is used in the Continue function.
+        /// </summary>
+        /// <param name="dc">The DialogContext.</param>
+        /// <returns>Boolean indicating whether the timout has expired.</returns>
```

Comment:
>timout [](start = 52, length = 6)

Typo.

Reply:
th only xplanation is th ky must not b working on my machin